wrapper/src/main/java/software/amazon/jdbc/plugin/OpenedConnectionTracker.java

```diff
@@ -50,16 +52,17 @@ public class OpenedConnectionTracker {
             invalidateThread.setDaemon(true);
             return invalidateThread;
           });
-  private static final ExecutorService abortConnectionExecutorService =
-      Executors.newCachedThreadPool(
-          r -> {
-            final Thread abortThread = new Thread(r);
-            abortThread.setDaemon(true);
-            return abortThread;
-          });
+  private static final Executor abortConnectionExecutor = new SynchronousExecutor();
```

Comment:
Just double checking, are tasks submitted to the executor daemon threads?

Reply:
Aborting connection is processed as a task in invalidateConnectionsExecutorService (that runs in a daemon thread) but the task uses SynchronousExecutor to abort each connection.